lti_consumer/plugin/views.py

```diff
@@ -415,3 +416,113 @@ def scores(self, request, *args, **kwargs):  # pylint: disable=unused-argument
             status=status.HTTP_201_CREATED,
             headers=headers
         )
+
+
+class LtiNrpsContextMembershipViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    LTI NRPS Context Membership Service endpoint.
+    """
+
+    # Custom permission classes for LTI APIs
+    authentication_classes = [Lti1p3ApiAuthentication]
+    permission_classes = [LtiNrpsContextMembershipsPermissions]
+
+    # Renderer classes to accept LTI NRPS content types
+    renderer_classes = [
+        MembershipResultRenderer,
+    ]
+
+    serializer_class = LtiNrpsContextMembershipSerializer
+
+    # Custom pagination class to support pagination via Link Header
+    pagination_class = LinkHeaderPagination
+
+    def parse_role_filter(self):
+        """
+        Parse role filter from query param and map it to edx role.
+        """
+        role_filter_param = self.request.query_params.get('role', None)
+
+        matched_role = None
+        if role_filter_param:
+            for edx_role, context_roles in LTI_1P3_CONTEXT_ROLE_MAP.items():
+                if role_filter_param in context_roles:
+                    matched_role = edx_role
+
+        return matched_role
+
+    def queryset_helper(self, kwarg_overrides):
```

Comment:
This might need to be changed depending on the direction of https://github.com/edx/edx-platform/pull/25843

Reply:
@giovannicimolin, yes this has been changed as the data API implementation got changed.